### ISSUE TITLE
Command to open main menu (start menu, application menu)

### ISSUE THREAD

**Issue by hiqsol** [lxqt/lxqt#1861]
I have disabled lxqt-globalkeysd because I have all keys configured in openbox.
I want to configure key to open main menu.
Is there a command to open main menu?
Previously in LXDE it was `lxpanelctl menu`

Thanks!

**Comment by palinek** [comment]
> Is there a command to open main menu?

No, currently there is not.

**Comment by ralf3u** [comment]
A command is not really necessary for the main menu. The solution is: Don't deactivate the shortcut for the main menu in the Shortcut Keys. Just set the shortcut for the main menu to an other shortcut, if you want to use the W-key also for something else in openbox, so for example instead of W, I use W+y for the main menu.

I would like to close this issue, but there is no Close-button.

**Comment by stefonarch** [comment]
Eventually this could be useful for wayland in future.

**Comment by stefonarch** [comment]
Is this possible for fancymenu @gfgit ?

**Comment by gfgit** [comment]
`lxpanelctl` uses X11 event to send special commands to lxpanel.

I don't think LXQt has something similar.
Anyway X11 event is not a good solution, DBus would be better.

**Comment by stefonarch** [comment]
On wayland  (until we are able to register  globalshortcuts) this could be useful as atm there is only a ugly hack possible to open the menu by shortcut:
```

#!/bin/bash
# Open Application menu on the top left from lxqt-panel
# Bind a key to it
wlrctl pointer move -2000 -1100 && sleep 0.2 && wlrctl pointer click

```

**Comment by gfgit** [comment]
Hi, I've just discovered this trick. @hiqsol it does not even require `lxqt-globalkeysd` running.
Since FancyMenu registers a global shortcut it will run it's DBus service which you can mess with like this:

```shell
qdbus org.kde.StatusNotifierWatcher /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated
```

Let me know if it works for you

**Comment by gfgit** [comment]
This of course can be done also for other panel plugins which register global actions

**Comment by stefonarch** [comment]
Works like a charm on wayland, nice catch!

**Comment by gfgit** [comment]
> Can this show and hide the task bar? I mean the part of the panel that shows window tabs, on which the user click to switch windows.

I don't think so. This is more of an hack than a proper solution. What's your usecase exactly?

**Comment by stefonarch** [comment]
@QiangF : "taskbar" is the "taskbar plugin" which shows the open windows, you mean a shortcut for showing the entire  panel if I understand well.

**Comment by gfgit** [comment]
You could write a new fake plugin which does not show anything but hooks up globalkeys to catch win key press and shows only the portion you want on the taskbar.
Very hacky but also cool to do.
At the moment unfortunately I'm busy with other things but if you want to try this I can give you some advices

**Comment by stefonarch** [comment]
I think @QiangF uses "taskbar" for the whole panel:

>Or I just want to look at the time and date on the bar with windows key down

**Comment by enihcam** [comment]
do we now have an official tutorial to do this (start app menu through CLI/dbus)? I'm using hyprland.

**Comment by stefonarch** [comment]
> do we now have an official tutorial to do this (start app menu through CLI/dbus)? I'm using hyprland.

It's still the mentioned 
`qdbus org.kde.StatusNotifierWatcher /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated` but afaik on many compositors it doesn't focus the search field and items automatically so it's only good with the mouse. There are workarounds for that, but ugly. Afaik in hyprland it works fine, in my config I've
```
#bindr = SUPER, Super_L, exec, qdbus org.kde.StatusNotifierWatcher /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated
bind  = Alt, F1, exec, qdbus org.kde.StatusNotifierWatcher /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated
```
but see yourself. `qdbus` is in `qt5-tools` aon arch, but there is also `qdbus6` which works the same.

**Comment by tsujan** [comment]
> but see yourself. qdbus is in qt5-tools aon arch, but there is also qdbus6 which works the same.

`qdbus6` should be used on Arch and its derivatives. Some distros may link `qdbus` to `qdbus6`, but Arch hasn't so far.

`qdbus` can be used too, but users don't need `qt5-tools` for LXQt (which is based on Qt6).

**Comment by tsujan** [comment]
And if there's any doubt about `qdbus6`, `dbus-send` can be used directly:

```
dbus-send --session --dest=org.kde.StatusNotifierWatcher --type=method_call /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated
```

BTW, the search field is focused on kwin_wayland.

**Comment by GuiFlam** [comment]
> And if there's any doubt about `qdbus6`, `dbus-send` can be used directly:
> 
> ```
> dbus-send --session --dest=org.kde.StatusNotifierWatcher --type=method_call /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated
> ```
> 
> BTW, the search field is focused on kwin_wayland.

Wow this is really cool

I am on labwc though, is there a way to make this command work on labwc? The menu appears, but the text field is not focused, which makes it kinda useless lol

Even if I click on the text field it does not focus if there are any other active text fields on the screen.

**Comment by tsujan** [comment]
> is there a way to make this command work on labwc?

We can't do anything about it. It's focused under some wlroots-based compositors, like Wayfire, and not under others. Some devs of the second group of compositors believe that the problem is in Qt, some not.

Anyway, kwin_wayland cares for Qt much more than other Wayland compositors; labwc comes in the second place, but its support is acceptable for most users (I patch wlroots for an issue it has with Qt though).

**Comment by GuiFlam** [comment]
> > is there a way to make this command work on labwc?
> 
> We can't do anything about it. I

Okay I just made this script, it is not the best way I would have imagined, but it works with a hacky ydotool setup.

Basically, for the focus to be on the textfield, you need to have beforehand clicked on the desktop bar. After doing this, the focus will be on the text bar in the fancy menu. I just virtualized this input using ydotool.

```
#!/bin/bash

if ! pgrep ydotoold > /dev/null; then
    sudo /home/mage/Files/ydotool/build/ydotoold &
    sleep 0.3
fi

# Move the mouse on the desktop bar
ydotool mousemove -x 0 -y 1184

# Click on the desktop bar to register focus                                              
ydotool/build/ydotool click 0xC0                                         
                                                                                          
# Open the panel: The focus is on the text field!                                         
dbus-send --session --dest=org.kde.StatusNotifierWatcher --type=method_call /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated &        
                                                                                          
# Move the mouse on the desktop bar                                                       
ydotool mousemove -x 0 -y 600                              
                                                                                          
sleep 0.3 # Delay needed for focus registration                                           
                                                                                          
# Move back the mouse to where it was                                                     
ydotool mousemove -x 0 -y -300 
```

**Comment by stefonarch** [comment]
Well, on the hacky said this works fine if you know where your menu is (and on  which screen eventually):

`wlrctl pointer move -2000 -1100 && sleep 0.2 && wlrctl pointer click` (for top left corner)

The `qdbus` command focuses in kwin, hyprland and wayfire only.
Edit: works on niri too now.

**Comment by Hate9** [comment]
Ok, so in case somebody else finds this and spends a ton of time searching like I did, the *current* way to do this is apparently `lxqt-qdbus openmenu`.

**Comment by stefonarch** [comment]
Yes, it was meant mainly for Wayland but will work (for fancy menu)  on X11 too, without using `lxqt-globalshortcutsd`.

**Comment by mdmahendri** [comment]
> Ok, so in case somebody else finds this and spends a ton of time searching like I did, the _current_ way to do this is apparently `lxqt-qdbus openmenu`.

thank you. i just installed fedora and lxqt labwc just a few days ago and need this feature. after installing `qt6-qttools` and registering through `rc.xml` this command works great (although i notice some delay compared to clicking it directly)

EDIT: i see, that command use `org.kde.StatusNotifierWatcher`

**Comment by stefonarch** [comment]
@mdmahendri quick OT question: in my VM with Fedora I see the "LXQt Settings" menu in fancy menu nearly empty, while all settings are under "Preferences". Is that also so on your installation?

**Comment by mdmahendri** [comment]
> [@mdmahendri](https://github.com/mdmahendri) quick OT question: in my VM with Fedora I see the "LXQt Settings" menu in fancy menu nearly empty, while all settings are under "Preferences". Is that also so on your installation?

yes, that is true. For me, only `Desktop` in LXQt Settings, while the others placed in Preferences

**Comment by stefonarch** [comment]
ok, thanks, will investigate then. It should be ordered like in `lxqt-config`.

**Comment by stefonarch** [comment]
@mdmahendri https://github.com/lxqt/lxqt/discussions/2816

**Comment by msqr1** [comment]
> Ok, so in case somebody else finds this and spends a ton of time searching like I did, the _current_ way to do this is apparently `lxqt-qdbus openmenu`.

This command is finally added in [LXQt 2.4.0](https://lxqt-project.org/release/2026/04/20/release-lxqt-2-4-0/) for Wayland. But since `lxqt-globalkeys` is not available on Wayland, if you want to use a keybind, you will have to set it on your compositor settings. Finally I don't need to use my mouse to click on it every time.

**Comment by ralf3u** [comment]
>[...], if you want to use a keybind, you will have to set it on your compositor settings.

Sorry to ask, but where do I find the compositor settings?

**Comment by stefonarch** [comment]
Session Settings → Wayland Settings → Compositor

>For the supported compositors, minimal default settings based on their default settings are shipped and will be installed in ~/.config/lxqt/wayland/ folder at the first start, except for kwin_wayland and labwc, where the default configurations are used. 

https://lxqt-project.org/wiki/Wayland-Session.html#compositors
